fanout/join_fanout.js

```diff
@@ -0,0 +1,183 @@
+    output_count: <Output Count Number>
+    rate: <Chain Fee Tokens Per VByte Number>
+  }
+
+  @returns
+  <EventEmitter Object>
+
+  @event 'end'
+  {
+    id: <Transaction Id Hex String>
+  }
+
+  // Sending the signatures for the open to the coordinator
+  @event 'publishing'
+  {}
+*/
+module.exports = (args, cbk) => {
+  const emitter = new EventEmitter();
+
+  asyncAuto({
+    // Import ECPair library
+    ecp: async () => (await import('ecpair')).ECPairFactory(tinysecp),
+
+    // Check arguments
+    validate: cbk => {
+      if (!isArray(args.inputs)) {
+        return cbk([400, 'ExpectedArrayOfUtxosToJoinFanout']);
+      }
+
+      if (!args.capacity) {
+        return cbk([400, 'ExpectedGroupFanoutOutputCapacityToJoinFanout']);
+      }
+
+      if (!args.coordinator) {
+        return cbk([400, 'ExpectedFanoutGroupCoordinatorIdToJoinFanout']);
+      }
+
+      if (!args.count) {
+        return cbk([400, 'ExpectedMembersCountToJoinFanout']);
+      }
+
+      if (!args.id) {
+        return cbk([400, 'ExpectedGroupIdToJoinFanout']);
+      }
+
+      if (!args.inputs.length) {
+        return cbk([400, 'ExpectedArrayOfUtxosToJoinFanout']);
+      }
+
+      if (!args.lnd) {
+        return cbk([400, 'ExpectedAuthenticatedLndToJoinFanout']);
+      }
+
+      if (!args.output_count || !isNumber(args.output_count)) {
+        return cbk([400, 'ExpectedOutputCountToJoinFanout']);
+      }
+
+      if (!args.rate) {
+        return cbk([400, 'ExpectedChainFeeRateToJoinFanout']);
+      }
+
+      return cbk();
+    },
+
+    // Confirm connected the partners
+    connected: ['validate', ({}, cbk) => {
+      return registerGroupConnected({
+        coordinator: args.coordinator,
+        count: args.count,
+        id: args.id,
+        lnd: args.lnd,
+      }, cbk);
+    }],
+
+    // Get the funding details for the fanout
+    getFundingInfo: ['connected', ({}, cbk) => {
+      return getFundingDetails({
+        capacity: args.capacity,
+        count: args.count,
+        inputs: args.inputs,
+        lnd: args.lnd,
+        output_count: args.output_count,
+        rate: args.rate,
+      },
+      cbk);
+    }],
+
+    // Register pending proposal and sign and fund
+    register: ['getFundingInfo', ({getFundingInfo}, cbk) => {
+      return registerPendingOpen({
+        capacity: args.capacity,
+        coordinator: args.coordinator,
+        change: getFundingInfo.change,
+        funding: getFundingInfo.funding,
+        group: args.id,
+        lnd: args.lnd,
+        overflow: getFundingInfo.overflow,
+        output_count: args.output_count,
+        pending: getFundingInfo.id,
+        service: serviceTypeRegisterPendingFanout,
+        utxos: getFundingInfo.utxos,
+      },
+      cbk);
+    }],
+
+    // Decode the unsigned PSBT
+    transaction: [
+      'ecp',
+      'getFundingInfo',
+      'register',
+      ({ecp, getFundingInfo, register}, cbk) =>
+    {
+      const funding = hexAsBuffer(getFundingInfo.funding);
```

Comment:
isn't funding here an array not a hex?

Reply:
actually nvm. should just remove vouts and funding